shell/components/PaginatedResourceTable.vue

```diff
@@ -86,7 +87,9 @@ export default defineComponent({
     ];
 
     if (this.fetchSecondaryResources) {
-      promises.push(this.fetchSecondaryResources({ canPaginate: this.canPaginate }));
+      const res = this.fetchSecondaryResources({ canPaginate: this.canPaginate });
+
+      promises.push(Array.isArray(res) ? res : [res]);
```

Comment:
By any chance do you mean to deconstruct the array before pushing? 

`promises.push(...(Array.isArray(res) ? res : [res]));` 

I don't think Promise.all() accepts nested arrays of promises.

Reply:
Good catch. That's a bug. There was another as well, as map of promises was the alternative

I reworked this in the end and now the response to `fetchSecondaryResources` will always be a single promise (the map was pointless as it would be just mapped into an array of promises). This might affect your PR